Sources/Mantle/Check.swift

```diff
@@ -368,14 +368,70 @@ extension TypeChecker where PhaseState == CheckPhaseState {
       self.extendEnvironment([(Name(name: name), patType)])
       return (.variable, type)
     case let .constructor(dataCon, synPats):
+      // Use the data constructor to locate back up the parent so we can
+      // retrieve its argument telescope.
+      guard let data = self.signature.lookupDefinition(dataCon) else {
+        fatalError()
```

Comment:
These should eventually be replaced with diagnostics, but that shouldn't block this merge.

Reply:
That one is a proper critical failure of ScopeCheck.  I'll blow through here and have much better primitives for identifying diagnostics versus proper failures.  I also want to try to do the Swift-y thing and crash showing the context of what we were doing and which expression we were type checking.

Reply:
Ooh, the PrettyStackTrace thing?